src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -499,11 +499,16 @@ private static Assembly GetRequiredLoadedAssembly(AssemblyKey assemblyKey)
             ?? throw new ArgumentException($"There is no loaded assembly with the name '{assemblyKey.AssemblyName}'.");
     }
 
-    private static void ClearCache(Type[]? _)
+    // don't point the MetadataUpdateHandlerAttribute at the DotNetDispatcher class, since the attribute has
+    // DynamicallyAccessedMemberTypes.All. This causes unnecessary trim warnings on the non-MetadataUpdateHandler methods.
+    internal static class MetadataUpdateHandler
     {
-        _cachedMethodsByAssembly.Clear();
-        _cachedMethodsByType.Clear();
-        _cachedConvertToTaskByType.Clear();
+        public static void ClearCache(Type[]? _)
+        {
+            _cachedMethodsByAssembly.Clear();
+            _cachedMethodsByType.Clear();
+            _cachedConvertToTaskByType.Clear();
```

Comment:
Unrelated to this change - but that's where I noticed this. These caches will effectively disable support for unloadable (collectible) assemblies - since they hold a string reference to the `Type`. Would be really nice if we didn't introduce more hurdles in this space in the new frameworks.

Reply:
@vitek-karas that's fair, but at the same time its a trade-off between perf and `correctness` on a very concrete scenario. Do we have a design/proposal on how to deal with these two concerns that we could apply across all frameworks?

In any case, that's something we should discuss outside of this PR, but if we come up with a reasonable design for it, I think we would incorporate it.